python/ray/ml/result.py

```diff
@@ -4,15 +4,31 @@
 
 
 class Result:
-    """The final result of a ML training run.
+    """The final result of a ML training run or a Tune trial.
 
-    This is the class produced by Trainer.fit(). It contains a checkpoint, which
-    can be used for resuming training and for creating a Predictor object. It also
-    contains a metrics object describing training metrics.
+    This is the class produced by Trainer.fit() or Tuner.fit() (through ResultGrid).
+    It contains a checkpoint, which can be used for resuming training and for
+    creating a Predictor object. It also contains a metrics object describing
+    training metrics. `status` and `error` are included so that non successful runs
```

Comment:
Are we also including "status"? (I think we should, for non-terminated trials)

Reply:
(This is sort of implementation detail and ideally shouldn't be part of API discussion, but...) Currently by the time TrialRunner is exited, there are only two kinds of trials: TERMINATED v.s. ERROR. Seems users are fine with it. So I am a little hesitant to change it as it's a pretty simple view. Assuming we don't want to change that, having `status` and `error` seems a bit redundant to me.

Reply:
Ok, can drop for now then.